apps/api-extractor/src/collector/Collector.ts

```diff
@@ -146,6 +156,69 @@ export class Collector {
     this._cachedOverloadIndexesByDeclaration = new Map<AstDeclaration, number>();
   }
 
+  /**
+   * Searches the provided package.json for dependencies and devDependencies that match the provided
+   * package names and/or RegExp patterns in `bundledPackages`.
+   * @param bundledPackages - The list of package names and/or RegExp patterns to search for in the package.json.
+   * @param packageJson - The package.json of the package being processed.
+   * @returns The set of matching package names.
+   */
+  private static _resolveBundledPackagePatterns(
```

Comment:
TODO: do we need to respect devDependencies here? Package's shouldn't be consuming other packages' dev dependencies.

A number of the test setups in this repo are using devDependencies with bundledPackages. This is easy enough to fix, but figured I would leave them alone for now and start a discussion.